File: Individual/Task1MyPharmacy/Task1MyPharmacy/src/tables/FlyweightFactory.cpp
#include "FlyweightFactory.h"
FlyweightProduct* FlyweightFactory::GetCartFlyweight(CartProduct& product, int amount)
{
	for (auto it : flyweightsArr)
	{
		if (it->id == product.id)
		{
			//it->amount = amount;
			return it;
		}
	}

	FlyweightProduct* flyw = new FlyweightProduct(product, amount);
	flyweightsArr.push_back(flyw);
	return flyw;
}

File: Individual/MyPharmacy/MyPharmacy/DAL/Repositories/FlyweightFactory.cs
using MyPharmacy.DAL.Modules.Impl;
using System.Collections.Generic;

namespace MyPharmacy.DAL.Repositories
{
    public class FlyweightFactory
    {
	    public Flyweight GetCartFlyweight(CartEntity product, int amount)
        {
			foreach (var it in flyweightsArr)
			{
				if (it.id == product.id)
				{
					return it;
				}
			}
			Flyweight flyw = new Flyweight(product, amount);
			flyweightsArr.Add(flyw);
			return flyw;
		}
	    public List<Flyweight> flyweightsArr = new List<Flyweight>();
    }
}
